README.md

```diff
@@ -68,9 +68,97 @@ ETags are not provided for or used consistently.
 
 Versioning is not supported at this time.
 
-# Setting it up
+# Docker
 
-## Building
+This project provides two Dockerfiles, one for building and one for running the application.
+
+**IMPORTANT: All commands in the sections that follow assume you are located in the root of the repository.**
+
+## The Builder Image
+
+The builder image is responsible for building the iRODS S3 API package. Before you can use it, you must build the image. To do that, run the following:
+```bash
+docker build -t irods-s3-api-builder -f irods_builder.Dockerfile .
+```
+
+With the builder image in hand, all that's left is to compile the source code for the S3 API project. The builder image is designed to compile code sitting on your machine. This is important because it gives you the ability to build any fork or branch of the project.
+
+Building the package requires mounting the project into the container at the appropriate location. The command you run should look similar to the one below. Don't forget to create the directory which will hold your package!
+```bash
+docker run -it --rm \
+    -v /path/to/irods_client_s3_api:/s3_api_source:ro \
+    -v /path/to/packages_directory:/packages_output \
+    irods-s3-api-builder
+```
+
+If everything succeeds, you will have a DEB package in the local directory you mapped to **/packages_output**.
+
+## The Runner Image
+
+The runner image is responsible for running the iRODS S3 API. Building the runner image requires the DEB package the iRODS S3 API to exist on the local machine. See the previous section for details on generating the package.
+
+To build the image, run the following command:
+```bash
+docker build -t irods-s3-api-runner -f irods_runner.Dockerfile /path/to/packages/directory
+```
+
+If all goes well, you will have a containerized iRODS S3 API server! You can verify this by checking the version information. Below is an example.
+```bash
+$ docker run -it --rm irods-s3-api-runner -v
+irods_s3_api v0.2.0-<build_sha>
```

Comment:
maybe we put a `<version>` here... 

or at least... 0.1.0 for now, we're not at 0.2.0 yet.

Reply:
Changed to `<version>`